x/evm/keeper/precompiles.go

```diff
@@ -31,3 +31,26 @@ func (k *Keeper) GetActivePrecompilesInstances(
 
 	return addresses, staticPrecompilesMap
 }
+
+// GetPrecompileInstance returns the address and instance of the static or dynamic precompile associated with the given address, or return nil if not found.
+func (k *Keeper) GetPrecompileInstance(
+	ctx sdktypes.Context,
+	address common.Address,
+) ([]common.Address, map[common.Address]vm.PrecompiledContract, bool) {
```

Comment:
Same question here, if we change the return types in `GetDynamicPrecompileInstance` and we return here an error, we can bubble up the error and be more informative on the failing reason